### ISSUE TITLE
CustomDNS should support other query types (CNAME, TXT, ...)

### ISSUE THREAD

**Issue by 0xERR0R** [0xERR0R/blocky#465]

**Comment by SuperSandro2000** [comment]
I added cname support in https://github.com/0xERR0R/blocky/pull/996

**Comment by BenMcH** [comment]
While the CNAME PR appears to be DOA, you can emulate CNAMEs for internal IPs using yaml anchor tags like this:

```yaml
customDNS:
  customTTL: 1h
  mapping:
    home-server.lan: &server 192.168.0.1
    app-1.lan: *server
    app-2.lan: *server
```

This of course has its downsides and doesn't solve for external domain names, but for my use-cases this works well enough. 

If I have time in the coming weeks/months, I may look at picking up where @SuperSandro2000 left off and add @0xERR0R's suggestion for psuedo-funcitons in the config file to see if we can get that merged. I know I'd prefer my apps to be CNAMEs even if this is technically functional.

**Comment by joryirving** [comment]
I saw that @BenMcH merged #1352 and I'm very thankful, but I wanted to know if there was a way to handle multiple zones in 0.24? I got one zone working wonderfully.

**Comment by kwitsch** [comment]
Shouldn't be a problem since according to [RFC1035](https://datatracker.ietf.org/doc/html/rfc1035) the master file format supports it. The commonly used name 'zone file' is a bit misleading since it supports more than just one zone.
